### PR TITLE
Fix konsole dbus call and  modify xsettingsd on kde as a fallback method

### PR DESCRIPTION
**Pull request by heddxh**
- Konsole seems don't split it sessions so it may can directly switch theme by calling `set_profile`
- Kde user may not install `kde-gtk-config`. In this case their gtk apps may be controlled by xsettingsd.

### CODE REVIEW

File: tests/test_plugins.py
Comment (l0drex) [inline tests/test_plugins.py:1]: Please revert these changes

File: yin_yang/__main__.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 from PySide6.QtWidgets import QSystemTrayIcon, QMenu
 from systemd import journal
 
+sys.path.append(str(Path(__file__).parent.parent))
```
Comment (l0drex) [inline yin_yang/__main__.py:15]: What is your intention here? A comment would be helpful

File: yin_yang/plugins/kvantum.py [hunk 2]
```diff
@@ -10,6 +12,20 @@ def __init__(self):
         self.theme_light = 'KvFlatLight'
         self.theme_dark = 'KvFlat'
 
+    def set_theme(self, theme: str):
```
Comment (l0drex) [inline yin_yang/plugins/kvantum.py:15]: This is nice, but the only benefit I can see of using dbus instead of the command line is that this works in flatpaks. This would be great, but we would need to convert the whole plugin from a command line based one to a standard plugin in order to get everything working inside a flatpak. This requires a bit more work, so maybe seperate this into a new pull request.

File: yin_yang/plugins/gtk.py [hunk 3]
```diff
@@ -79,7 +80,23 @@ def set_theme(self, theme: str):
...
+                logger.warning('xsettingsd not available')
+            with open(xsettingsd_conf_path, 'r') as f:
+                lines = f.readlines()
+                for i, line in enumerate(lines):
+                    if line.startswith('Net/ThemeName'):
+                        lines[i] = f'Net/ThemeName "{theme}"\n'
+                        break
+            with open(xsettingsd_conf_path, 'w') as f:
+                f.writelines(lines)
+            subprocess.run(['killall', '-HUP', 'xsettingsd'])
```
Comment (l0drex) [inline yin_yang/plugins/gtk.py:97]: I guess the intent here is to restart xsettingsd. Is there a nicer way of doing this, e.g. with `systemctl`?

File: yin_yang/plugins/konsole.py [hunk 4]
```diff
@@ -74,6 +74,7 @@ def set_mode(self, dark: bool) -> bool:
             logger.debug(f'Changing profile in konsole session {proc_id}')
             set_profile(f'org.kde.konsole-{proc_id}', profile)
 
+        set_profile('org.kde.konsole', profile) # konsole may don't have session dbus like above
```
Comment (l0drex) [inline yin_yang/plugins/konsole.py:77]: If possible, it might be a better idea to check if the method above was successful and then execute this alternative.

**Review by l0drex — CHANGES_REQUESTED** [review]
Thanks for your work! I have some minor requests, other then that it looks good!

**Comment by heddxh** [comment]
Sorry I forgot push to my fork will also affect my pr.
I will clean my code!

File: tests/test_plugins.py
Comment (heddxh) [reply tests/test_plugins.py:1]: OK, I forgot...

File: yin_yang/__main__.py (hunk 1 above)
Comment (heddxh) [reply yin_yang/__main__.py:15]: Import local model in VScode need add project location to path and I forgot to revert it

File: yin_yang/plugins/kvantum.py (hunk 2 above)
Comment (heddxh) [reply yin_yang/plugins/kvantum.py:15]: OK, I will revert it and make a new pr when I an ready

File: yin_yang/plugins/gtk.py (hunk 3 above)
Comment (heddxh) [reply yin_yang/plugins/gtk.py:97]: Its [document](https://codeberg.org/derat/xsettingsd) say I can sedding `HUG` signal to reload config but I think `systemctl restart` also works. Which one do you think is better？

File: yin_yang/plugins/konsole.py (hunk 4 above)
Comment (heddxh) [reply yin_yang/plugins/konsole.py:77]: OK, I finally figure it out. When user checked option in konsole.s setting -> Common -> Running all konsole windows in one process, konole's dbus service will not seperate. So yes I should try both way.

Comment (heddxh) [reply yin_yang/plugins/konsole.py:77]: When that option is checked, I can only toggle current konsole window to change profile but if I open new window or tab, it still remain using old profile. Only if I close all konsole window and then open new one, it will use the new default profile

Comment (l0drex) [reply yin_yang/plugins/konsole.py:77]: Ok, so we need to check the config file and decide from there

**Review by l0drex — APPROVED** [review]
Looks good, thanks!
